### PR TITLE
Fix that config.bind(key, None) unbind the key

### PR DESCRIPTION
**Pull request by Jan-PieterBaert**
This should fix my issue #4948

### CODE REVIEW

**Comment by Jan-PieterBaert** [comment]
I will take a look at it

**Comment by Jan-PieterBaert** [comment]
I will have a look at it

**Comment by Jan-PieterBaert** [comment]
@The-Compiler is there anything else I should add?

**Comment by The-Compiler** [comment]
Can you please take a look at the failing CI output?

- [Coverage](https://travis-ci.org/qutebrowser/qutebrowser/jobs/567051481)
- [pylint](https://travis-ci.org/qutebrowser/qutebrowser/jobs/567051483)
- [flake8](https://travis-ci.org/qutebrowser/qutebrowser/jobs/567051484)
- [mypy](https://travis-ci.org/qutebrowser/qutebrowser/jobs/567051485)

The mypy one seems a bit tricky - in `__init__` the type of `bindings` is wrong, so mypy thinks a command in a binding can never be `None`. I think it should go away if you change the type of the command there from `str` to `typing.Optional[str]` - if that doesn't help, feel free to ignore it and I'll fix it up while merging.

**Comment by Jan-PieterBaert** [comment]
@The-Compiler tests seem to work now

File: tests/unit/config/test_configfiles.py [hunk 1]
```diff
@@ -1000,6 +1000,28 @@ def test_binding_options_hidden(self):
         assert 'bindings.default' not in text
         assert 'bindings.commands' not in text
 
+    def test_unbind(self):
+        opt = configdata.Option(
+            name='opt', typ=configtypes.Int(), default='def',
+            backends=[usertypes.Backend.QtWebEngine], raw_backends=None,
+            description='Hello World')
+        options = [(None, opt, 'val')]
```
Comment (The-Compiler) [inline tests/unit/config/test_configfiles.py]: This doesn't seem to be required for what you actually want to test - you're just interested in `config.unbind(...)` (and maybe a `config.bind`) working correctly, so I think you could just pass `options=[]` to `ConfigPyWriter` for this test to simplify it.

File: tests/unit/config/test_configfiles.py [hunk 2]
```diff
@@ -1000,6 +1000,28 @@ def test_binding_options_hidden(self):
...
+        bindings = {'normal': {',x': None},
+                    'caret': {',y': 'message-info caret', ',z': None}}
+
+        writer = configfiles.ConfigPyWriter(options, bindings, commented=False)
+        lines = list(writer._gen_lines())
+
+        assert "# Autogenerated config.py" in lines
+        assert "# config.load_autoconfig()" in lines
+        assert "c.opt = 'val'" in lines
+        assert "# Bindings for normal mode" in lines
```
Comment (The-Compiler) [inline tests/unit/config/test_configfiles.py]: Same as above - you're not actually interested in those things for this particular test, so let's remove them and simplify the test.

**Comment by Jan-PieterBaert** [comment]
The test was a copy of another test and I forgot to remove the unrelated lines

**Comment by The-Compiler** [comment]
Thanks!

**Comment by The-Compiler** [comment]
I just noticed that the old `config.bind(key, None)` lines actually worked before v1.8.0 (before 9ed6aaaa1646fad0710ba28bf12bd8eb9e011a72). So I guess some old configs might be around which worked fine until v1.8.0 but now show a (rather confusing) exception.

I now re-added that functionality with a deprecation warning in 2033ed1955b642cbcd2bc4ea88d361c340fb3d38.

**Pull request merged**